frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/BasePushBranchAction.java

```diff
@@ -0,0 +1,66 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import java.util.Collections;
+
+import com.intellij.dvcs.push.ui.VcsPushDialog;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.Project;
+import git4idea.GitLocalBranch;
+import git4idea.push.GitPushSource;
+import git4idea.repo.GitRepository;
+import io.vavr.collection.List;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public abstract class BasePushBranchAction extends DumbAwareAction {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("frontendActions");
+
+  protected final List<SyncToRemoteStatus.Relation> PUSH_ENABLING_STATUSES = List.of(
+      SyncToRemoteStatus.Relation.Ahead,
+      SyncToRemoteStatus.Relation.Diverged,
+      SyncToRemoteStatus.Relation.Untracked);
+
+  /**
+   * Bear in mind that {@link AnAction#beforeActionPerformedUpdate} is called before each action.
+   * (For more details check {@link com.intellij.openapi.actionSystem.ex.ActionUtil} as well.)
+   * The {@link AnActionEvent} argument passed to before-called {@link AnAction#update} is the same one that is passed here.
+   * This gives us certainty that all checks from actions' update implementations will be performed
+   * and all data available via data datakeys in those {@code update} implementations will still do be available
+   * in {@link BasePushBranchAction#actionPerformed} implementations.
+   */
+  @Override
+  @UIEffect
+  public abstract void actionPerformed(AnActionEvent anActionEvent);
+
+  @UIEffect
+  public void doPush(Project project, GitRepository preselectedRepository, String branchName) {
+    @Nullable
+    GitLocalBranch localBranch = preselectedRepository.getBranches().findLocalBranch(branchName);
+
+    if (localBranch != null) {
+      java.util.List<GitRepository> selectedRepositories = Collections.singletonList(preselectedRepository);
+      new VcsPushDialog(project,
+          /* allRepositories */ selectedRepositories,
+          /* preselectedRepositories */ selectedRepositories,
+          /* currentRepo */ null,
```

Comment:
Hmmm why is this one `null`? Or rather, what would it mean if it weren't null?

Reply:
After a look into `com.intellij.dvcs.push.PushController#PushController` I came to the following conclusions:

Push dialog presents commits of selected branches that are to be pushed.
A multi-repo push with lots of commits may take some time to load all commits.
`PushController` controls the order in which commits of branches (aka `RepositoryNode`s) will be loaded.
`currentRepo` specifies the current repository which will have higher priority (is loaded as first).

As our actions (push current/selected) push a single repo only we can depend on `preselectedRepositories` (containing current repo only). 

Does it answer your question?

Reply:
Ok! Pls drop comment in the code with (a possibly abridged version of) your response